src/r_float.rs

```diff
@@ -35,10 +38,12 @@ impl RFloat {
         Self(NonZeroValue::new_unchecked(Value::new(val)))
     }
 
+    /// TODO: ?????
     /// Create a new `RFloat` from an `f64.`
     ///
     /// Returns `Ok(RFloat)` if `n` requires a high precision float, otherwise
     /// returns `Err(Fixnum)`.
+    #[cfg(ruby_use_flonum)]
     pub fn from_f64(n: f64) -> Result<Self, Flonum> {
```

Comment:
The semantics of this are now a bit confusing given that flonum support is optional. My initial feeling is we should deprecate this, and encourage `Float::from_f64` instead. Thoughts?

Reply:
I think it should probably be `Option<RFloat>` when flonum is not supported.

Reply:
I feel like that may violate the "principle of least surprise" because it could a compilation error when trying to compile on 32bit.

Although weird at a glance, I'm thinking `Result<Self, Self>` would make the most sense since `rb_float_new` will always return an `RFloat` when `USE_FLONUM=0`.

Reply:
Oh, yeah, `Result<Self, Self>` makes sense.

Reply:
Done!